test/test.js

```diff
@@ -0,0 +1,60 @@
+const Application = require('spectron').Application;
+const path = require('path');
+const chai = require('chai');
+const chaiAsPromised = require('chai-as-promised');
+const loginPage = require('./pageObjects/login.page.js');
+
+var LoginPage = new loginPage();
+
+var electronPath = path.join(__dirname, '..', 'node_modules', '.bin', 'electron');
+
+if (process.platform === 'win32') {
+    electronPath += '.cmd';
+}
+
+var appPath = path.join(__dirname, '..');
+
+var app = new Application({
+            path: electronPath,
+            args: [appPath]
```

Comment:
If we launch the application like this, that is, not from inside a test harness which would allow injection changes, we need another way to specify that the code should use the mocked services, such as an environment variable or a command line parameter to our app. Without this, we won't be able to write a test that logs in and performs operations in the app.

If this is too much work to do now, I can create another ticket for that work to split it out.

Reply:
If it's ok to you I'd prefer to tackle this in another story.